test/websocket.test.js

```diff
@@ -5,6 +5,7 @@
 const assert = require('assert');
 const crypto = require('crypto');
 const https = require('https');
+const { URL } = require('url');
```

Comment:
Doesn't work on Node.js 4.

Reply:
Node 4 doesn't have the URL instance either, so I've opted for adding in some horrible if statements in the test suite :/